apps/emqx_oracle/src/emqx_oracle.erl

```diff
@@ -146,7 +146,14 @@ on_query(
     Type = query,
     {NameOrSQL2, Data} = proc_sql_params(TypeOrKey, NameOrSQL, Params, State),
     Res = on_sql_query(InstId, PoolName, Type, ?SYNC_QUERY_MODE, NameOrSQL2, Data),
-    handle_result(Res).
+    case Res of
+        {error, {recoverable_error, _Error}} ->
+            Res;
+        {error, {unrecoverable_error, {_RetCode, _Reason}}} ->
+            Res;
```

Comment:
Maybe we could add those clauses to `handle_result`?

Reply:
it could be...